### PR TITLE
Add post-inference hooks to Endpoint create/edit

### PR DESCRIPTION
**Pull request by phil-scale**
# Pull Request Summary
Companion PR to https://github.com/scaleapi/models/pull/3213

## Clubhouse Ticket
[[ch418719]](https://app.clubhouse.io/scaleai/story/418719)

### CODE REVIEW

File: launch/hooks.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from enum import Enum
+
+
+class PostInferenceHooks(str, Enum):
+    INSIGHT = "INSIGHT"
```
Comment (syandroo) [inline launch/hooks.py:5]: do python standards prefer upper case? if not we can just make this lowercase just like the equals check

File: launch/client.py [hunk 2]
```diff
@@ -448,6 +449,7 @@ def create_model_endpoint(
         per_worker: int = 1,
         gpu_type: Optional[str] = None,
         endpoint_type: str = "sync",
+        post_inference_hooks: Optional[List[PostInferenceHooks]] = None,
```
Comment (syandroo) [inline launch/client.py:452]: should we make this empty array just in case someone breaks the default in the server side?

**Review by syandroo — APPROVED** [review]
lgtm two small comments

File: launch/hooks.py (hunk 1 above)
Comment (phil-scale) [reply launch/hooks.py:5]: sure we can do lower-case

File: launch/client.py (hunk 2 above)
Comment (phil-scale) [reply launch/client.py:452]: Good article on why default should be None not empty list: https://satran.in/b/python--dangerous-default-value-as-argument

Tl;dr is that if you have the default as `[]`, it's considered unsafe since the interpreter could treat the list as the same reference between **different calls** to the method.

Comment (phil-scale) [reply launch/client.py:452]: (Actually I'm not that pro and didn't know that) - `pylint` caught it for me 😆
